picard/ui/metadatabox/__init__.py

```diff
@@ -265,41 +272,107 @@ def keyPressEvent(self, event):
         else:
             super().keyPressEvent(event)
 
+    def get_selected_tags(self, items):
+        result = TagDiff()
+        for item in items:
+            tag, value = self.get_row_info(item.row())
+            col = item.column()
+            result.add(
+                tag=tag,
+                old=value[self.COLUMN_ORIG] if col == self.COLUMN_ORIG else None,
+                new=value[self.COLUMN_NEW] if col == self.COLUMN_NEW else None,
+                removable=self.tag_diff.status[tag] != TagStatus.NOTREMOVABLE,
+                readonly=self.tag_diff.status[tag] == TagStatus.READONLY
+            )
+
+        result.update_tag_names()
+        return result
+
+    def get_row_info(self, row):
+        tag = self.tag_diff.tag_names[row]
+        value = {}
+        value[self.COLUMN_ORIG] = self.tag_diff.old[tag]
+        value[self.COLUMN_NEW] = self.tag_diff.new[tag]
+
+        return tag, value
+
     def _copy_value(self):
-        item = self.currentItem()
-        if item:
-            column = item.column()
-            tag = self.tag_diff.tag_names[item.row()]
-            value = None
-            if column == self.COLUMN_ORIG:
-                value = self.tag_diff.old[tag]
-            elif column == self.COLUMN_NEW:
-                value = self.tag_diff.new[tag]
+        if len(self.tracks) > 1 or len(self.files) > 1:
+            # We cannot copy from multiple files at the same time, clear the clipboard to avoid confusion
+            log.debug("Unable to copy when multiple tracks or files are selected, clearing the clipboard")
```

Comment:
Why this restriction? In the current code copying is possible.

I think we shouldn't restrict this here. When copying multiple selected tags it should instead ignore the tags that differ between the files, but copy all tags that are the same.

When copying a single value we could either retain the current behavior (which copies an empty string in case of differing values) or not copy anything.

Reply:
Whilst we don't stop people copying with multiple files selected currently, it does crash Picard (at least in my experience!) So rather than trying to amalgamate a bugfix for that into this PR I figured adding this failsafe for now and fix the underlying issue in a different PR.

Reply:
It's working fine for me. The only crash I'm aware of is the one you already fixed in https://github.com/metabrainz/picard/pull/2609 .

So we should be good to allow this. At a minimum as part of this PR I would expect that the copying of a single value keeps working as before. Having multiple tags work across multiple files we could do in a separate PR, if it complicates things too much.

If there are still some crash conditions in existing code, I agree we can also fix them separate from this PR.

Reply:
> It's working fine for me. The only crash I'm aware of is the one you already fixed in https://github.com/metabrainz/picard/pull/2609

Wow that is embarrassing - I'm rediscovering bugs that I've already fixed! 🫣